server/requirements.txt

```diff
@@ -1,76 +1,57 @@
+opentelemetry-exporter-otlp==1.21.0 ; python_version >= "3.9" and python_version < "4.0"
+opentelemetry-instrumentation-grpc==0.42b0 ; python_version >= "3.9" and python_version < "4.0"
+opentelemetry-instrumentation==0.42b0 ; python_version >= "3.9" and python_version < "4.0"
+opentelemetry-proto==1.21.0 ; python_version >= "3.9" and python_version < "4.0"
+opentelemetry-sdk==1.21.0 ; python_version >= "3.9" and python_version < "4.0"
+opentelemetry-semantic-conventions==0.42b0 ; python_version >= "3.9" and python_version < "4.0"
+packaging==23.2 ; python_version >= "3.9" and python_version < "4.0"
+protobuf==4.25.1 ; python_version >= "3.9" and python_version < "4.0"
+pyyaml==6.0.1 ; python_version >= "3.9" and python_version < "4.0"
+regex==2023.10.3 ; python_version >= "3.9" and python_version < "4.0"
 requests==2.31.0 ; python_version >= "3.9" and python_version < "4.0"
 safetensors==0.3.1 ; python_version >= "3.9" and python_version < "4.0"
-scipy
 sentencepiece==0.1.99 ; python_version >= "3.9" and python_version < "4.0"
-setuptools==68.0.0 ; python_version >= "3.9" and python_version < "4.0"
-six==1.16.0 ; python_version >= "3.9" and python_version < "4.0"
+setuptools==69.0.2 ; python_version >= "3.9" and python_version < "4.0"
 sympy==1.12 ; python_version >= "3.9" and python_version < "4.0"
-texttable==1.6.7 ; python_version >= "3.9" and python_version < "4.0"
+tiktoken==0.5.2 ; python_version >= "3.9" and python_version < "4.0"
 tokenizers==0.15.0 ; python_version >= "3.9" and python_version < "4.0"
-torch==2.0.1 ; python_version >= "3.9" and python_version < "4.0"
-tqdm==4.65.0 ; python_version >= "3.9" and python_version < "4.0"
+torch==2.1.1+cu118 ; python_version >= "3.9" and python_version < "4.0"
+tqdm==4.66.1 ; python_version >= "3.9" and python_version < "4.0"
 transformers==4.35.2 ; python_version >= "3.9" and python_version < "4.0"
+triton==2.1.0 ; python_version >= "3.9" and python_version < "4.0"
 typer==0.6.1 ; python_version >= "3.9" and python_version < "4.0"
-typing-extensions==4.7.1 ; python_version >= "3.9" and python_version < "4.0"
-tzdata==2023.3 ; python_version >= "3.9" and python_version < "4.0"
-urllib3==1.26.12 ; python_version >= "3.9" and python_version < "4.0"
+typing-extensions==4.8.0 ; python_version >= "3.9" and python_version < "4.0"
+urllib3==2.1.0 ; python_version >= "3.9" and python_version < "4.0"
 win32-setctime==1.1.0 ; python_version >= "3.9" and python_version < "4.0" and sys_platform == "win32"
-wrapt==1.15.0 ; python_version >= "3.9" and python_version < "4.0"
-xxhash==3.2.0 ; python_version >= "3.9" and python_version < "4.0"
-yarl==1.9.2 ; python_version >= "3.9" and python_version < "4.0"
-peft==0.4.0 ; python_version >= "3.9" and python_version < "4.0"
-boto3==1.28.34 ; python_version >= "3.9" and python_version < "4.0"
+wrapt==1.16.0 ; python_version >= "3.9" and python_version < "4.0"
+zipp==3.17.0 ; python_version >= "3.9" and python_version < "4.0"
```

Comment:
Looks like we lost `peft==0.4.0` somehow. Can you add this to the `pyproject.toml`?